server/src/internalClusterTest/java/org/opensearch/index/store/CorruptedFileIT.java

```diff
@@ -149,8 +152,86 @@ protected Collection<Class<? extends Plugin>> nodePlugins() {
     }
 
     /**
-     * Tests that we can actually recover from a corruption on the primary given that we have replica shards around.
+     * Tests that we cannot recover from a corruption on the primary that is actively indexing, even if it has replica.
      */
+    public void testCorruptFileDuringIndexing() throws Exception {
```

Comment:
One of the ways of reproducing this:
1. Create a node with overridden send behavior for TransportWriteAction through `MockTransportService`. You could add random delays or repro precisely via a latch.
2. Create an index with a primary and replica. Ensure replica gets initialized on node that executes delays.
3. Issue indexing operations and corrupt primary shard as done in the current test.
4. Release the latch in step 1.
5. Ensure that cluster health is green.


How to create a node with an overridden action:

Reply:
Apologies for the delay, but I haven't been able to get an integration test that both deterministically fails without the fix and passes with the fix. I've got the fix polished up with more targeted unit tests. Let me know what you think.